src/nvim/autocmd.c

```diff
@@ -190,7 +190,18 @@ static void aupat_show(AutoPat *ap, event_T event, int previous_group)
     if (got_int) {
       return;
     }
-    msg_outtrans((char_u *)aucmd_exec_to_string(ac, ac->exec));
+
+    char *exec_to_string = aucmd_exec_to_string(ac, ac->exec);
+    if (ac->desc != NULL) {
+      size_t msglen = 100;
+      char *msg = (char *)xmallocz(msglen);
+      snprintf(msg, msglen, "%s [%s]", exec_to_string, ac->desc);
+      msg_outtrans((char_u *)msg);
+      xfree(msg);
+    } else {
+      msg_outtrans((char_u *)exec_to_string);
+    }
+    xfree(exec_to_string);
```

Comment:
If `ac->exec.type` is CALLABLE_EX then you are freeing the autocommand's command string (which is returned from `aucmd_exec_to_string`. `aucmd_exec_to_string` should `strdup` the command string.